### PR TITLE
GodMode changes whole file intead of only the lines that have changed #4237

### LINKED ISSUE #4237: GodMode changes whole file instead of only the lines that have changed
This is caused by spacing differences in the browsers in different computers.

### PR DESCRIPTION
**Pull request by thyageshm**
fixes #4237

### CODE REVIEW

**Comment by thyageshm** [comment]
@LowWeiLin could you review this? 

FYI, I will be uploading a huge list of files (godmode run on all html files) soon.

**Comment by wkurniawan07** [comment]
Be careful when running God mode on admin pages. They have lots of unexpected wildcards.

**Comment by thyageshm** [comment]
Hi! Thanks for the heads up :-) I have fixed as many of those as possible on my system. I have requested wei lin to help me identify any that I have missed

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -803,6 +804,7 @@ private static String processPageSourceForGodMode(String content) {
                 .replace(TestProperties.inst().TEST_UNREG_ACCOUNT, "${test.unreg}")
                 .replace(Config.SUPPORT_EMAIL, "${support.email}")
                 // today's date
+                .replace(TimeHelper.formatDate(now).replace("/", "&#x2f;"), "{*}")
```
Comment (LowWeiLin) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: What is this doing?

**Comment by LowWeiLin** [comment]
When running tests, some UI tests fail (may not be persistent failure) but eclipse shows there is no diff. This is not supposed to happen right?

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 1 above)
Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: If this has something to do with sanitization, on account of #4244, let's put this on hold first.
On another note, @bamboo3250 it'd be great if you can do it ASAP.

**Comment by LowWeiLin** [comment]
There are about 30 UI tests failing. (Can't see why for some as there is no diff)
teammates-thy.appspot.com is hardcoded in some tests.

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 1 above)
Comment (LowWeiLin) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I get this failure, probably related to this line of code?
image

Comment (bamboo3250) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Okay, I'll do it now.

**Comment by damithc** [comment]
The page is beautified after detecting the error but before generating the diff, the the beautifying might be rectifying what caused the error.

File: src/test/java/teammates/test/pageobjects/AppPage.java (hunk 1 above)
Comment (bamboo3250) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: That issue has been fixed. Please make change accordingly.

Comment (wkurniawan07) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Good job @bamboo3250. Now I believe this line change is no longer necessary.

Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: ok will remove it

**Comment by thyageshm** [comment]
I think the I generated the html files before #4244 was fixed and so GodMode did not work properly in some cases where the sanitisation made the comparison rules to fail. I will regenerate them now that the redundant sanitization has been removed. That could also be the cause for failure without any diff as @damithc rightly pointed out

**Comment by wkurniawan07** [comment]
Also, not too important, but you may also want to use this opportunity to change the indentation to 4 spaces (currently it's 3 which is odd; pun unintended :p) and remove the closing tag in void elements (like `<br/>`), both of which are not compliant to our style guide.

**Comment by damithc** [comment]
Do those separately if possible. It's never a good idea to do multiple things in a PR unless they absolutely must be done together.

**Comment by wkurniawan07** [comment]
Ah yes do it separately _before this PR_ if you want to. Or else you'll need to regenerate again afterwards.

**Pull request merged**